app/jobs/scheduled/clean_up_unmatched_ips.rb

```diff
@@ -11,7 +11,7 @@ def execute(args)
 
       # remove old unmatched IP addresses
       ScreenedIpAddress.where(action_type: ScreenedIpAddress.actions[:block])
-                       .where("last_match_at < ?", last_match_threshold)
+                       .where("last_match_at < ? OR last_match_at IS NULL AND created_at < ?", last_match_threshold, last_match_threshold)
```

Comment:
I can make it use paranthesizes, fairly certain order of operations will process the AND first though. If it makes you feel better, I'll definitely add them.